usecases/mu/mpc/public.go

```diff
@@ -0,0 +1,545 @@
+		}
+	}
+
+	return powerPerPhase, nil
+}
+
+// Scenario 2
+
+// get the total feed in energy
+//
+//   - negative values are used for production
+//
+// possible errors:
+//   - ErrMissingData if the id is not available
+//   - and others
+func (e *MPC) EnergyConsumed() (float64, error) {
+	if e.acEnergyConsumed == nil {
+		return 0, api.ErrMissingData
+	}
+
+	return e.getMeasurementDataForId(e.acEnergyConsumed)
+}
+
+// get the total feed in energy
+//
+//   - negative values are used for production
+//
+// possible errors:
+//   - ErrMissingData if the id is not available
+//   - and others
+func (e *MPC) EnergyProduced() (float64, error) {
+	if e.acEnergyProduced == nil {
+		return 0, api.ErrMissingData
+	}
+
+	return e.getMeasurementDataForId(e.acEnergyProduced)
+}
+
+// Scenario 3
+
+// get the momentary phase specific current consumption or production
+//
+//   - positive values are used for consumption
+//   - negative values are used for production
+//
+// possible errors:
+//   - ErrMissingData if the id is not available
+//   - and others
+func (e *MPC) CurrentPerPhase() ([]float64, error) {
+	currentPerPhase := make([]float64, 0)
+
+	for _, id := range e.acCurrent {
+		if id != nil {
+			current, err := e.getMeasurementDataForId(id)
+			if err != nil {
+				return nil, err
+			}
+			currentPerPhase = append(currentPerPhase, current)
+		}
+	}
+
+	return currentPerPhase, nil
+}
+
+// Scenario 4
+
+// get the phase specific voltage details
+//
+// possible errors:
+//   - ErrMissingData if the id is not available
+//   - and others
+func (e *MPC) VoltagePerPhase() ([]float64, error) {
+	voltagePerPhase := make([]float64, 0)
+
+	for _, id := range e.acVoltage {
+		if id != nil {
+			voltage, err := e.getMeasurementDataForId(id)
+			if err != nil {
+				return nil, err
+			}
+			voltagePerPhase = append(voltagePerPhase, voltage)
+		}
+	}
+
+	return voltagePerPhase, nil
+}
+
+// Scenario 5
+
+// get frequency
+//
+// possible errors:
+//   - ErrMissingData if the id is not available
+//   - and others
+func (e *MPC) Frequency() (float64, error) {
+	if e.acFrequency == nil {
+		return 0, api.ErrMissingData
+	}
+
+	return e.getMeasurementDataForId(e.acFrequency)
+}
+
+// ------------------------- Setters ------------------------- //
+
+// use MPC.Update to update the measurement data
+// use it like this:
+//
+//	mpc.Update(
+//	  mpc.UpdateDataPowerTotal(1000, nil, nil),
+//	  mpc.UpdateDataPowerPhaseA(500, nil, nil),
+//	  ...
+//	)
+//
+// possible errors:
+//   - ErrMissingData if the id is not available
+//   - and others
+func (e *MPC) Update(updateData ...usecaseapi.UpdateMeasurementData) error {
+	measurements, err := server.NewMeasurement(e.LocalEntity)
+	if err != nil {
+		return err
+	}
+
+	measurementDataForIds := make([]api.MeasurementDataForID, 0)
+
+	for _, measurementDataForId := range updateData {
+		if !measurementDataForId.Supported() {
+			return measurementDataForId.NotSupportedError()
+		} else {
```

Comment:
The `else` statement is not needed here, as we return in the other case.

So this code could be simplified to:

```golang
if !measurementDataForId.Supported() {
  return measurementDataForId.NotSupportedError()
}

measurementDataForIds = append(measurementDataForIds, measurementDataForId.MeasurementData())
```